iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/helplets.inl

```diff
@@ -168,17 +168,25 @@ inline bool doesEndWithPathSeparator(const string<StringCapacity>& name) noexcep
     return false;
 }
 
-template <typename F, typename T>
-inline constexpr T from(const F e IOX_MAYBE_UNUSED) noexcept
+template <typename SourceType, typename DestinationType>
+inline constexpr DestinationType from(const SourceType e)
 {
-    static_assert(always_false_v<F> && always_false_v<T>, "Conversion for the specified types is not implemented!\
-    Please specialize `template <typename F, typename T> constexpr T from(const F) noexcept`!");
+    return From<SourceType, DestinationType>::fromImpl(e);
 }
 
-template <typename T, typename F>
-inline constexpr T into(const F e) noexcept
+
+template <typename SourceType, typename DestinationType>
+inline DestinationType From<SourceType, DestinationType>::fromImpl(const SourceType& value IOX_MAYBE_UNUSED)
+{
+    static_assert(always_false_v<SourceType> && always_false_v<DestinationType>,
+                  "Conversion for the specified types is not implemented!\
+    Please specialize `template <typename SourceType, typename DestinationType> constexpr DestinationType from(const SourceType) noexcept`!");
```

Comment:
This nice statement is not correct. Assume the struct is renamed into `Into::intoImpl` the user has to add a specialization for this struct so that it can be used from the free function `into`.
See:
```cxx
template<>
struct Into<SomeOriginType, MyFancyType> {
     static MyFancyType fromImpl(const SomeOriginType& value) {}
}
```

It would be also a nice idea to add an simple example to the doxygen documentation of the `Into` struct - an example to convert `float` into `string` for instance.

Reply:
@elfenpiff Thanks your the hint! Adapted the wording to the current state. Let's keep any future renaming out of this PR and do it on a follow-up.